File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        strs = s.lower()
        left = 0
        right = len(s) - 1
        while left < right:
            while not strs[left].isalnum() and left < right:
                left += 1
            while not strs[right].isalnum() and left < right:
                right -= 1
            if strs[left] != strs[right]:
                return False
            left += 1
            right -= 1
        return True

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isBalanced(TreeNode* root) {
        return dfs(root) != -1;
    }
    int dfs(TreeNode* root) {
        if (!root) return 0;
        int l = dfs(root->left);
        int r = dfs(root->right);
        if (l == -1 or r == -1) {
            return -1;
        }
        if (abs(l - r) <= 1) {
            return max(l,r) + 1;
        } else {
            return -1;
        }
    }
};
